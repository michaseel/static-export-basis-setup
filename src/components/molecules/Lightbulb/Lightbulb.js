import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  background-color: darkblue;
  padding: 70px;
`;

const Bulp = styled.div`
  border-radius: 50%;
  width: 200px;
  height: 200px;
  background-color: ${({ isOn, color }) => (isOn ? color : 'gray')};
  box-shadow: 
    inset -3px 2px 40px -2px white,
    0 0 80px 10px ${({ isOn, color }) => (isOn ? color : 'transparent')};
  transition: background-color .3s, box-shadow .1s;
  margin: 0 auto;
  opacity: .9;
  
  :after {
    position: relative;
    display: block;
    width: 80px;
    top: 20px;
    transform: rotate(330deg);
    margin-left: 30px;
    height: 50px;
    border-radius: 80%;
    background-image:
      linear-gradient(to top, rgba(255, 255, 255, 0), rgba(255, 255, 255, 0.4));
    background-size: 50px 50px;
    content: '';
  }
`;

const Socket = styled.div`
  position: relative;
  background: linear-gradient(
    to bottom,
    #999,
    #999 50%,
    #666 50%,
    #666
  );
  background-size: 100% 25px;
  width: 110px;
  height: 70px;
  margin: -15px auto 0;
`;

const Connector = styled.div`
  box-shadow: inset 0 0 6px -2px white;
  background-color: #333;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  margin: -30px auto 0;
`;

export default class Lightbulb extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isOn: false,
    };
  }

  static propTypes = {
    /** The color of the lightbulb */
    color: PropTypes.string,
  };

  toggle = () => {
    this.setState({ isOn: !this.state.isOn });
  };

  static defaultProps = {
    color: '#f8e897',
  };

  render() {
    return (
      <Wrapper>
        <Bulp onClick={this.toggle} color={this.props.color} isOn={this.state.isOn}>{this.props.stars}</Bulp>
        <Socket />
        { this.state.isOn ? <Socket/> : <Connector /> }
        <Connector />
      </Wrapper>
    );
  }
}
