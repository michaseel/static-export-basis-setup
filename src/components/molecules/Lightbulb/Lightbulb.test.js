import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Lightbulb from './Lightbulb';

it('renders default Lightbulb snapshot', () => {
  const tree = renderer.create(<Lightbulb />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders yellow Lightbulb snapshot', () => {
  const tree = renderer.create(<Lightbulb color="yellow" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders red Lightbulb snapshot', () => {
  const tree = renderer.create(<Lightbulb color="red" />).toJSON();
  expect(tree).toMatchSnapshot();
});
