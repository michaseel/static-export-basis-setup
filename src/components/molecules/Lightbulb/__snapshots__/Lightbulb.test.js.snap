// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Lightbulb snapshot 1`] = `
.c0 {
  background-color: darkblue;
  padding: 70px;
}

.c1 {
  border-radius: 50%;
  width: 200px;
  height: 200px;
  background-color: gray;
  box-shadow: inset -3px 2px 40px -2px white, 0 0 80px 10px transparent;
  -webkit-transition: background-color .3s,box-shadow .1s;
  transition: background-color .3s,box-shadow .1s;
  margin: 0 auto;
  opacity: .9;
}

.c1:after {
  position: relative;
  display: block;
  width: 80px;
  top: 20px;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  margin-left: 30px;
  height: 50px;
  border-radius: 80%;
  background-image: linear-gradient(to top,rgba(255,255,255,0),rgba(255,255,255,0.4));
  background-size: 50px 50px;
  content: '';
}

.c2 {
  position: relative;
  background: linear-gradient( to bottom, #999, #999 50%, #666 50%, #666 );
  background-size: 100% 25px;
  width: 110px;
  height: 70px;
  margin: -15px auto 0;
}

.c3 {
  box-shadow: inset 0 0 6px -2px white;
  background-color: #333;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  margin: -30px auto 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="#f8e897"
    onClick={[Function]}
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;

exports[`renders red Lightbulb snapshot 1`] = `
.c0 {
  background-color: darkblue;
  padding: 70px;
}

.c1 {
  border-radius: 50%;
  width: 200px;
  height: 200px;
  background-color: gray;
  box-shadow: inset -3px 2px 40px -2px white, 0 0 80px 10px transparent;
  -webkit-transition: background-color .3s,box-shadow .1s;
  transition: background-color .3s,box-shadow .1s;
  margin: 0 auto;
  opacity: .9;
}

.c1:after {
  position: relative;
  display: block;
  width: 80px;
  top: 20px;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  margin-left: 30px;
  height: 50px;
  border-radius: 80%;
  background-image: linear-gradient(to top,rgba(255,255,255,0),rgba(255,255,255,0.4));
  background-size: 50px 50px;
  content: '';
}

.c2 {
  position: relative;
  background: linear-gradient( to bottom, #999, #999 50%, #666 50%, #666 );
  background-size: 100% 25px;
  width: 110px;
  height: 70px;
  margin: -15px auto 0;
}

.c3 {
  box-shadow: inset 0 0 6px -2px white;
  background-color: #333;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  margin: -30px auto 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="red"
    onClick={[Function]}
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;

exports[`renders yellow Lightbulb snapshot 1`] = `
.c0 {
  background-color: darkblue;
  padding: 70px;
}

.c1 {
  border-radius: 50%;
  width: 200px;
  height: 200px;
  background-color: gray;
  box-shadow: inset -3px 2px 40px -2px white, 0 0 80px 10px transparent;
  -webkit-transition: background-color .3s,box-shadow .1s;
  transition: background-color .3s,box-shadow .1s;
  margin: 0 auto;
  opacity: .9;
}

.c1:after {
  position: relative;
  display: block;
  width: 80px;
  top: 20px;
  -webkit-transform: rotate(330deg);
  -ms-transform: rotate(330deg);
  transform: rotate(330deg);
  margin-left: 30px;
  height: 50px;
  border-radius: 80%;
  background-image: linear-gradient(to top,rgba(255,255,255,0),rgba(255,255,255,0.4));
  background-size: 50px 50px;
  content: '';
}

.c2 {
  position: relative;
  background: linear-gradient( to bottom, #999, #999 50%, #666 50%, #666 );
  background-size: 100% 25px;
  width: 110px;
  height: 70px;
  margin: -15px auto 0;
}

.c3 {
  box-shadow: inset 0 0 6px -2px white;
  background-color: #333;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  margin: -30px auto 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    color="yellow"
    onClick={[Function]}
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
</div>
`;
